### PR TITLE
Add support for CloudFront functions

### PR DESCRIPTION
**Pull request by alexjurkiewicz**
## what

This follows the naming of Terraform's AWS provider. Lambda@Edge
functions are "lambda functions", whereas CloudFront functions are
simply "functions".

## references
* https://github.com/hashicorp/terraform-provider-aws/releases/tag/v3.41.0
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_distribution#function-association
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_function

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -462,6 +470,14 @@ resource "aws_cloudfront_distribution" "default" {
           lambda_arn   = lambda_function_association.value.lambda_arn
         }
       }
+
+      dynamic "function_association" {
+        for_each = lookup(ordered_cache_behavior.value, "function_association", [])
```
Comment (alexjurkiewicz) [inline main.tf:475]: this is a lookup for backwards compatibility

**Comment by alexjurkiewicz** [comment]
I've tested this by deploying my own distribution with an attached function 👍

**Comment by nitrocode** [comment]
By adding this, shouldn't we also update the aws provider pin to > 3.41.0 ?

It would be nice to link the docs for the `function_association` resource argument.

**Comment by alexjurkiewicz** [comment]
Thanks for the feedback, done both!

**Review by nitrocode — APPROVED** [review]

**Comment by alexjurkiewicz** [comment]
thank you!

**Pull request merged**